File: Programming_Languages/asgn2/python_code/task3_java/Refrigerator.java
package assg2task3;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author kenchan
 */
public class Refrigerator extends SmartFixture {
    protected double temperature;
    public Refrigerator(String Name, double Weight, int Position, double Battery_capacity, double Initial_temperature){
        super(Name, Weight, Position, Battery_capacity);
        this.temperature = Initial_temperature;
    }
    public void SetTemperature(double New_temperature){
        this.temperature = New_temperature;
        System.out.println(this.name + " was set to " + New_temperature + " degrees Celsius.");
    }
}

File: Programming_Languages/asgn2/python_code/task3_python/Refrigerator.py
# this is the python refrigerator code

import SmartFixture
import Attendant
import Room

class Refrigerator(SmartFixture):
    def __init__(self,Name,Weight,Position,Battery_capacity,Initial_temperature):
        super(Refrigerator,self).__init__(Name,Weight,Position,Battery_capacity)
        self._temperature= Initial_temperature

    def SetTemperature(self,New_temperature):
        self._temperature=New_temperature
        print ('%s was set to %.2f degrees Celsius.' % self._name, New_temperature)
